public/static/app/perfdiscovery/PerformanceDiscoveryCtrl.js

```diff
@@ -1,29 +1,70 @@
-mciModule.controller('PerformanceDiscoveryCtrl', function($scope) {
+mciModule.controller('PerformanceDiscoveryCtrl', function(
+  $scope, $q, ApiTaskdata, ApiV1, PerfDiscoveryService, PERF_DISCOVERY
+) {
   var vm = this;
 
   vm.revisionSelect = {
-    options: [
-      {id: 1, name: '4ef5fx',},
-      {id: 2, name: 'v52fe3',},
-      {id: 3, name: 'd43ap0',},
-    ],
+    options: [],
+    selected: null,
   }
 
-  vm.revisionSelect.selected = vm.revisionSelect.options[0]
-
   vm.tagSelect = {
-    options: [
-      {id: 1, name: 'v3.6.1',},
-      {id: 2, name: 'v3.6.0',},
-      {id: 3, name: 'v3.5.2',},
-    ],
+    options: [],
+    selected: null,
   }
 
-  vm.tagSelect.selected = vm.tagSelect.options[0]
+  // TODO use globally defined project id
+  var projectId = 'sys-perf'
+
+  ApiV1.getProjectVersions(projectId).then(function(data) {
+    vm.versions = data.data.versions
+    vm.revisionSelect.options = _.map(data.data.versions, function(d, i) {
+      return {id: i, name: d.revision}
+    })
+    vm.revisionSelect.selected = vm.revisionSelect.options[0]
+
+    vm.tagSelect.options = _.map(data.data.versions, function(d, i) {
+      return {id: i, name: d.revision}
```

Comment:
For any versions where extractTasks returns an empty array (meaning the version didn't run), we should ideally remove them from the list. Would that be difficult to do?

Reply:
Good point. It should be possible. Let's create a bug ticket for this.